src/useObserver.ts

```diff
@@ -36,11 +36,13 @@ export function useObserver<T>(
                 forceUpdate()
             }
         })
+        scheduleCleanupOfReactionIfNotCommitted(reaction.current)
     }
 
     useDebugValue(reaction, printDebugValue)
 
     useEffect(() => {
+        recordReactionAsCommitted(reaction.current!)
```

Comment:
See comments in the conversation: if `reaction.isDisposed` yields `true`, the grace period has passed and we should re-run and re-render the component, to get a fresh reaction object.